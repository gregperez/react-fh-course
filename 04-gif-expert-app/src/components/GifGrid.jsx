import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);
    
    const getImages = async () => {
        const newImagens = await getGifs( category );
        setImages(newImagens);
    };

    useEffect( () => {
        getImages();
    }, []);


    return (
        <>
            <h3>{ category }</h3>

            <ol>
                { 
                    images.map( ({ id, title }) => (
                    <li key={ id }>{ title }</li>
                    ))
                }
            </ol>
        </>
    )
}