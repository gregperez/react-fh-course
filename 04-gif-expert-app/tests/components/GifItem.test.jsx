import { render } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {

    const title = 'Saitama';
    const url   = 'http://site.imagen.jpg';
    
    test('debe de hacer match conn el snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url } />)
        expect( container ).toMatchSnapshot();
    });
    
});
