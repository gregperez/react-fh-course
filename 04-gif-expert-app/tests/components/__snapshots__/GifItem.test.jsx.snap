// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <GifItem /> debe de hacer match conn el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://site.imagen.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
